File: experiments_dataset_vul_detect_cwe_classification/dataset/vul_dataset/c/non-vul/code_before13.c
// Fro cwe-snippets, snippets_90/non-compliant/C/cwe-0476/deref_after_check_01.c

static void func()
{
    {
        int *intPointer = NULL;
        if (intPointer == NULL)
        {
            printLine("pointer is NULL");
        }
    }
}

File: experiments_dataset_vul_detect_cwe_classification/dataset/vul_dataset/java/non-vul/code_before13.java
// From cwe-snippets, snippets_90/non-compliant/Java/cwe-0484/basic_03.java

private void func() throws Throwable
{
    if (5 != 5)
    {
        IO.writeLine("Benign, fixed string");
    }
    else
    {

        int intRandom = (new SecureRandom()).nextInt(3);
        String stringValue;

        switch (intRandom)
        {
        case 1:
            stringValue = "one";
            break;
        case 2:
            stringValue = "two";
            break; 
        default:
            stringValue = "Default";
            break;
        }

        IO.writeLine(stringValue);

    }
}